tests/test_root_model.py

```diff
@@ -43,10 +43,6 @@ class InnerModel(BaseModel):
 def test_root_model_specialized(root_type, root_value, dump_value):
     Model = RootModel[root_type]
 
-    assert Model.__pydantic_core_schema__['type'] == 'model'
-    assert Model.__pydantic_core_schema__['root_model'] is True
-    assert Model.__pydantic_core_schema__['custom_init'] is False
```

Comment:
Removing these since they're making assumptions that we don't necessarily promise and we shouldn't be testing implementation details like `root_model=True`.

Reply:
I think it's maybe worth keeping the `custom_init` one as there's a little trick to get this to be False, the others I agree can be removed.